src/QirRuntime/lib/TracingContext/TracingContext.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.

```

Comment:
Given that the resources estimation tool is still experimental, how come this context isn't? How is it supposed to be used?

Reply:
This context uses a linker seem to avoid virtual methods so LLVM could fold the loops with gate counts and it's not related to TracingSimulator. It should be moved into experimental.